lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -24,6 +24,29 @@ using namespace mlir;
 using namespace circt;
 using namespace seq;
 
+/// Determine the integer value of a constant operand.
+static std::optional<APInt> getConstantInt(Attribute operand) {
```

Comment:
It feels like these are useful in general (and likely copied from combfolds). Would it make sense to put them in a shared utility header?

Reply:
I didn't find any such op in the comb folds yet. But we should definitely add such a header to reduce code duplication.